File: 788 - Rotated Digits.js
/**
 * @param {number} N
 * @return {number}
 */
var rotatedDigits = function (N) {
    let res = 0;
    for (let i = 1; i <= N; i++) {
        if (checkRotation(i)) {
            res++;
        }
    }
    return res;
};

function checkRotation(n) {
    let flag = false, valid = true;
    while (n > 0 && valid) {
        let digit = n % 10;
        n = Math.floor(n / 10);
        switch (digit) {
            case 4:
            case 7:
            case 3:
                valid = false;
                flag = false;
                break;
            case 2:
            case 5:
            case 6:
            case 9:
                flag = true;
                break;
        }
    }
    return flag;
}

File: 788 - Rotated Digits.java
class Solution {
    public int rotatedDigits(int N) {
        int res = 0;
        for (int i = 1; i <= N; i++) {
            if (checkRotation(i)) {
                res++;
            }
        }
        return res;
    }

    public boolean checkRotation(int n) {
        int t = n;
        boolean flag = false, valid = true;
        int digit;
        while (t > 0 && valid) {
            digit = t % 10;
            t /= 10;
            switch(digit) {
                case 3, 4, 7:
                    valid = false;
                    flag = false;
                    break;
                case 2, 5, 6, 9:
                    flag = true;
                    break;
            }
        }
        return flag;
    }
}